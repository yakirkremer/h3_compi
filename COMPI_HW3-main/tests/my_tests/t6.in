void foo(int a, byte v,bool d){
    int num1;
    {
        int num2;
        bool asd;
    }
    int num2;
    while (true){
        int num3;
        int num4;
        int num5;
    }
    int num4;
}

void foo2(int a, byte v,bool d){
    int num1;
    {
        int num2;
        bool asd;
    }
    int num2;
    while (true){
        int num3;
        int num4;
        int num5;
    }
    int num4;
}

void main(){
    return;
}