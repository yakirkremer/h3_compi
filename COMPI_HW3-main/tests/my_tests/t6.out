---end scope---
num2 INT 1
asd BOOL 2
---end scope---
num3 INT 2
num4 INT 3
num5 INT 4
---end scope---
---end scope---
a INT -1
v BYTE -2
d BOOL -3
num1 INT 0
num2 INT 1
num4 INT 2
---end scope---
num2 INT 1
asd BOOL 2
---end scope---
num3 INT 2
num4 INT 3
num5 INT 4
---end scope---
---end scope---
a INT -1
v BYTE -2
d BOOL -3
num1 INT 0
num2 INT 1
num4 INT 2
---end scope---
---end scope---
print (STRING)->VOID 0
printi (INT)->VOID 0
foo (INT,BYTE,BOOL)->VOID 0
foo2 (INT,BYTE,BOOL)->VOID 0
main ()->VOID 0
